Source/HTTP/httpcall_request.cpp

```diff
@@ -108,6 +107,11 @@ try
     }
     RETURN_IF_PERFORM_CALLED(call);
 
+    if (!Compression::Available() && level != HCCompressionLevel::None)
+    {
+        return E_NOT_SUPPORTED;
+    }
+
```

Comment:
Any concern about this being a runtime failure as opposed to knowing its not supported at compile time?

Reply:
As discussed offline, the client should be free to just ignore the error, in the end they still want the request to be made compression or not.